File: Creational/Builder/JavaScript/js/ChickenMealBuilder.js
// js/ChickenMealBuilder.js
import { MealBuilder } from './MealBuilder.js';
import { Meal } from './Meal.js';

export class ChickenMealBuilder extends MealBuilder {
    constructor() {
        super();
        this.meal = new Meal();
    }

    buildMainItem() {
        this.meal.setMainItem('Chicken Sandwich');
    }

    buildSideItem() {
        this.meal.setSideItem('Nuggets');
    }

    buildDrink() {
        this.meal.setDrink('Sprite');
    }

    buildToy() {
        this.meal.setToy('Toy Dinosaur');
    }

    getMeal() {
        return this.meal;
    }
}

File: Creational/Builder/Java/src/main/java/com/fastfood/ChickenMealBuilder.java
package main.java.com.fastfood;

public class ChickenMealBuilder implements MealBuilder {
    private Meal meal;

    public ChickenMealBuilder() {
        this.meal = new Meal();
    }

    @Override
    public void buildMainItem() {
        meal.setMainItem("Chicken Sandwich");
    }

    @Override
    public void buildSideItem() {
        meal.setSideItem("Nuggets");
    }

    @Override
    public void buildDrink() {
        meal.setDrink("Sprite");
    }

    @Override
    public void buildToy() {
        meal.setToy("Toy Dinosaur");
    }

    @Override
    public Meal getMeal() {
        return this.meal;
    }
}
